bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/service/AbstractWatchService.java

```diff
@@ -77,7 +85,9 @@ protected void initializeWatchService() {
                 if (watchSubDirectories()) {
                     watchService = FileSystems.getDefault().newWatchService();
 
-                    Files.walkFileTree(toWatch, new SimpleFileVisitor<Path>() {
+                    // walk through all folders and follow symlinks
+                    Files.walkFileTree(toWatch, Sets.newHashSet(FileVisitOption.FOLLOW_LINKS), MAX_DEPTH, 
```

Comment:
* How is the max depth related to "adding follow symlink option"?
* Why do we need to restrict the depth of the abstract watch service? Isn't it a specific usage if the depth is limited to a specific value?